pkg/cloudprovider/providers/gce/gce_instances.go

```diff
@@ -407,87 +393,70 @@ func (gce *GCECloud) AddAliasToInstance(nodeName types.NodeName, alias *net.IPNe
 	remaining := len(names)
 
 	nodeInstancePrefix := gce.nodeInstancePrefix
 	for _, name := range names {
 		name = canonicalizeInstanceName(name)
 		if !strings.HasPrefix(name, gce.nodeInstancePrefix) {
-			glog.Warningf("instance '%s' does not conform to prefix '%s', removing filter", name, gce.nodeInstancePrefix)
+			glog.Warningf("Instance %q does not conform to prefix %q, removing filter", name, gce.nodeInstancePrefix)
 			nodeInstancePrefix = ""
 		}
-		instances[name] = nil
+		found[name] = nil
 	}
 
 	for _, zone := range gce.managedZones {
 		if remaining == 0 {
 			break
 		}
-
-		pageToken := ""
-		page := 0
-		for ; page == 0 || (pageToken != "" && page < maxPages); page++ {
-			listCall := gce.service.Instances.List(gce.projectID, zone)
-
-			if nodeInstancePrefix != "" {
-				// Add the filter for hosts
-				listCall = listCall.Filter("name eq " + nodeInstancePrefix + ".*")
+		instances, err := gce.c.Instances().List(context.Background(), zone, filter.Regexp("name", nodeInstancePrefix+".*"))
+		if err != nil {
+			return nil, err
+		}
+		for _, inst := range instances {
+			if remaining == 0 {
+				break
 			}
-
-			// TODO(zmerlynn): Internal bug 29524655
-			// listCall = listCall.Fields("items(name,id,disks,machineType)")
-			if pageToken != "" {
-				listCall.PageToken(pageToken)
+			if _, ok := found[inst.Name]; !ok {
+				continue
 			}
-
-			res, err := listCall.Do()
-			if err != nil {
-				return nil, err
+			if found[inst.Name] != nil {
+				glog.Errorf("Instance name %q was duplicated (in zone %q and %q)", inst.Name, zone, found[inst.Name].Zone)
+				continue
 			}
-			pageToken = res.NextPageToken
-			for _, i := range res.Items {
-				name := i.Name
-				if _, ok := instances[name]; !ok {
-					continue
-				}
-
-				instance := &gceInstance{
-					Zone:  zone,
-					Name:  name,
-					ID:    i.Id,
-					Disks: i.Disks,
-					Type:  lastComponent(i.MachineType),
-				}
-				instances[name] = instance
-				remaining--
+			found[inst.Name] = &gceInstance{
+				Zone:  zone,
+				Name:  inst.Name,
+				ID:    inst.Id,
+				Disks: inst.Disks,
+				Type:  lastComponent(inst.MachineType),
 			}
+			remaining--
 		}
-		if page >= maxPages {
-			glog.Errorf("getInstancesByNames exceeded maxPages=%d for Instances.List: truncating.", maxPages)
+	}
+
+	if remaining > 0 {
+		var failed []string
+		for k := range found {
+			failed = append(failed, k)
 		}
+		glog.Errorf("Failed to retrieve instances: %v", failed)
+		return nil, cloudprovider.InstanceNotFound
 	}
 
-	instanceArray := make([]*gceInstance, len(names))
-	for i, name := range names {
+	instanceArray := make([]*gceInstance, 0, len(names))
+	for _, name := range names {
 		name = canonicalizeInstanceName(name)
-		instance := instances[name]
-		if instance == nil {
-			glog.Errorf("Failed to retrieve instance: %q", name)
-			return nil, cloudprovider.InstanceNotFound
-		}
-		instanceArray[i] = instances[name]
+		instanceArray = append(instanceArray, found[name])
```

Comment:
Instead of touching `names` and `found` to create `instanceArray`, can we just iterate `found`?